clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -552,11 +552,12 @@ public void close(Duration timeout) {
             log.debug("Waiting for the I/O thread to exit. Hard shutdown in {} ms.", deltaMs);
         }
         try {
-            // Wait for the thread to be joined.
-            thread.join();
-
-            AppInfoParser.unregisterAppInfo(JMX_PREFIX, clientId, metrics);
-
+            // KAFKA-9330: close() can be called by AdminClient thread when it invokes
```

Comment:
nit: I think there's no need to reference the JIRA. The explanation seems clear enough without any additional context.

Reply:
Done.